ocaml/xapi/storage_migrate.ml

```diff
@@ -423,6 +426,20 @@ let start' ~task ~dbg ~sr ~vdi ~dp ~url ~dest =
 
   let id = State.mirror_id_of (sr,local_vdi.vdi) in
 
+  debug "Adding to active local mirrors before sending: id=%s" id;
+  let alm = State.Send_state.({
+      url;
+      dest_sr=dest;
+      remote_dp="";
```

Comment:
I don't like the idea of using empty strings as a signal that we've not finished setting up - I'd rather use option types here. Perhaps the values that are set in the 'Replacing active local mirrors' section below could all be a record that's an optional field in Send_state?

Reply:
Yes, I have putted them into an `option` object, and I think the tapdev should be an individual `option` object.